src/hotspot/share/gc/parallel/psPromotionLAB.hpp

```diff
@@ -118,9 +118,9 @@ class PSOldPromotionLAB : public PSPromotionLAB {
     // assert(_state != flushed, "Sanity");
     assert(_start_array != NULL, "Sanity");
     HeapWord* obj = top();
-    HeapWord* new_top = obj + size;
-    // The 'new_top>obj' check is needed to detect overflow of obj+size.
-    if (new_top > obj && new_top <= end()) {
+    // Pointer overflow check is needed here.
+    if (end() >= obj && size <= (size_t) (end() - obj)) {
```

Comment:
`top() <= end()` is always true by design and construction, so the
`end() >= obj` test is unnecessary and undesirable. An assertion could
be added, but there are lots of places where that relation is assumed
to hold and that don't bother asserting, so I wouldn't bother here
either.

Reply:
Instead of `(size_t)(end() - obj)` use `pointer_delta(end(), obj)`.

Reply:
Cool. Will update soon. Thanks.